components/automate-ui/src/app/pages/project/rules/project-rules.component.ts

```diff
@@ -149,33 +141,30 @@ export class ProjectRulesComponent implements OnInit, OnDestroy {
         this.attributes = attributes;
       });
 
-    // handle rule creation/update success
-    combineLatest([
-      this.store.select(createStatus),
-      this.store.select(updateStatus)
-    ]).pipe(
+    this.store.select(createStatus).pipe(
       takeUntil(this.isDestroyed),
-      filter(([cStatus, uStatus]) => this.saving && (!pending(cStatus) || !pending(uStatus))))
-      .subscribe(([cStatus, uStatus]) => {
-        this.saving = false;
-        if (cStatus === EntityStatus.loadingSuccess || uStatus === EntityStatus.loadingSuccess) {
-          this.closePage();
-        }
-      });
+      filter(status => this.saving && status === EntityStatus.loadingSuccess))
+      .subscribe(() => this.closePage());
+
+    this.store.select(updateStatus).pipe(
+      takeUntil(this.isDestroyed),
+      filter(status => this.saving && status === EntityStatus.loadingSuccess))
+      .subscribe(() => this.closePage());
 
-    // handle rule creation failure
     combineLatest([
       this.store.select(createStatus),
       this.store.select(createError)
     ]).pipe(
       takeUntil(this.isDestroyed),
-      filter(([state, error]) => state === EntityStatus.loadingFailure && !isNil(error)))
+      filter(([state, error]) =>
+        this.saving && state === EntityStatus.loadingFailure && !isNil(error)))
```

Comment:
Bug fix: Key piece missing was the `this.saving` inclusion in the filter so upon subsequent reopening of the full page dialog this would have triggered showing the error immediately.